### PR TITLE
Showcase: convert ShwLogo to `gts`

### PR DESCRIPTION
**Pull request by shleewhite**
### :pushpin: Summary

If merged, this PR would convert the ShwLogo::DesignSystem component to `.gts`.

### :link: External links

Jira ticket: [HDS-3835](https://hashicorp.atlassian.net/browse/HDS-3835)

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3835]: https://hashicorp.atlassian.net/browse/HDS-3835?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: showcase/.eslintrc.js [hunk 1]
```diff
@@ -26,10 +26,26 @@ module.exports = {
         functions: 'only-multiline',
       },
     ],
-    // ember doesn't support template only components that have type signatures so we need to use empty component classes
-    'ember/no-empty-glimmer-component-classes': 'off',
   },
   overrides: [
+    //  gts files
+    {
+      files: ['**/*.gts'],
```
Comment (shleewhite) [inline showcase/.eslintrc.js:33]: Note: I based the overrides on the HDS+ repo https://github.com/hashicorp/hds-plus/blob/5bbd63af89f581f52ed1983d777992920e3e7c60/showcase/.eslintrc.js#L42

File: showcase/app/components/shw/logo/design-system.gts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      class="shw-logo-hds__logo"
+      width="42"
+      height="42"
+      viewBox="0 0 42 42"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      aria-hidden="true"
+    >
+      <defs>
+        <linearGradient
```
Comment (shleewhite) [inline showcase/app/components/shw/logo/design-system.gts:20]: Note: without changing the linting rules, this file would always error saying that linearGradient isn't defined.

**Comment by Dhaulagiri** [comment]
> upgrade eslint-plugin-ember a major version

It's my preference but I also think a requirement that we extract this into a separate PR and do this for all the packages at once. Otherwise the new major will resolve for everything and lead to the issues you're seeing getting the component tests to pass.

**Review by aklkv — APPROVED** [review]
Looks great 👍🏻 This components is something we might think to refactor, I have been thinking that having non modified svg is overkill to have it's own component @didoo but this would have to be in the future

**Comment by shleewhite** [comment]
> @shleewhite is there something that needs to be done in this branch to get the tests passed?

Not on this branch, I need to merge the eslint upgrade pr first and then I think this pr will be unblocked https://github.com/hashicorp/design-system/pull/2427

**Review by zamoore — APPROVED** [review]
Nice!

**Pull request merged**